src/Bundle/Maker/Factory/ObjectDefaultPropertiesGuesser.php

```diff
@@ -62,4 +63,53 @@ public function supports(MakeFactoryData $makeFactoryData): bool
     {
         return !$makeFactoryData->isPersisted();
     }
+
+    private function shouldAddPropertyToFactory(MakeFactoryQuery $makeFactoryQuery, \ReflectionProperty $property): bool
+    {
+        // if option "--all-fields" was passed
+        if ($makeFactoryQuery->isAllFields()) {
+            return true;
+        }
+
+        // if property is inside constructor, check if it has a default value
+        if ($constructorParameter = $this->getConstructorParameterForProperty($property)) {
+            return !$constructorParameter->isDefaultValueAvailable();
+        }
+
+        // if the property has a default value, we should not add it to the factory
+        if ($property->hasDefaultValue()) {
+            return false;
+        }
+
+        // if property has type, we need to add it to the factory
+        return $property->hasType();
+    }
+
+    private function getPropertyType(\ReflectionProperty $property): string|null
+    {
+        if (!$property->hasType()) {
+            $type = $this->getConstructorParameterForProperty($property)?->getType();
+        } else {
+            $type = $property->getType();
+        }
+
+        if (!$type instanceof \ReflectionNamedType) {
+            return null;
+        }
+
+        return $type->getName();
+    }
+
+    private function getConstructorParameterForProperty(\ReflectionProperty $property): \ReflectionParameter|null
+    {
+        if ($constructor = $property->getDeclaringClass()->getConstructor()) {
+            foreach ($constructor->getParameters() as $parameter) {
+                if ($parameter->getName() === $property->getName()) {
+                    return $parameter;
+                }
+            }
+        }
+
+        return null;
+    }
```

Comment:
this method it a little bit biased because we rely on the fact the constructor's parameters have the same name than the property.

I think this could be accepted because it is a pattern widely used